Keys, Map, Compass found on Desert Collosus' hands go to Spirit Temple.

leggettc18 opened the pull request.

leggettc18 committed:
Keys, Map, and Compass found on Desert Collosus' hands count for Spirit Temple.

briaguya0 commented on soh/src/code/z_parameter.c:
```diff
@@ -1661,24 +1661,46 @@ u8 Item_Give(GlobalContext* globalCtx, u8 item) {
+            }
+            // if we get a boss/big key, dungeon map, or compass in Desert Colossus, it is
+            // for Spirit Temple.
+            if (gSaveContext.mapIndex == 11) {
```
having it in this order means that you'll set the flag for mapindex 11 and mapindex 6

you probably want
```
if (gSaveContext.mapIndex == 13) {
 [...]
} else if (gSaveContext.mapIndex == 11) {
  [...]
} else {
 [...]
}
```

leggettc18 replied:
Good catch, fixed in 1fe85c9

leggettc18 committed:
Fixes order of if statements.

Previously it would've set the Boss Key, Dungeon Map, or Compass flags for both Desert Collosus and Spirit Temple. Now it will only do Spirit Temple.
